### ISSUE TITLE
Wrong vram detected (4GB instead of 6GB)

### ISSUE THREAD

**Issue by laurent-clouet** [laurent-clouet/sheepit-client#88]
Sheepit's client does not detect the right amount of ram on the gpu, its maximum detection is 4GB even when the actual value is over (for example 6GB for geforce 980).

Vram of gpu are limited to 4294967295, it's actually the max value for 32bits integer but a gpu could have more.

CUDA api native api use an unsigned long to handle memory size but it look like it's limited to 32bits when it's use on sheepit (maybe jna limitation ? )

Tested on ubuntu 16.04 with cuda 7.5.

**Comment by sommd** [comment]
The GTX 980 has [4GB of VRAM](http://www.geforce.com/hardware/desktop-gpus/geforce-gtx-980/specifications). Do you mean the GTX 980 Ti?

**Comment by laurent-clouet** [comment]
Yes i meant 980TI.

**Comment by sommd** [comment]
Could be that it's using the 32-bit lib? [```cuDeviceTotalMem```](https://www.cs.cmu.edu/afs/cs/academic/class/15668-s11/www/cuda-doc/html/group__CUDA__DEVICE_gc6a0d6551335a3780f9f3c967a0fde5d.html) takes ```size_t``` which is 32 bits when compiling with -m32 and 64 bit when compiling with -m64.

**Comment by laurent-clouet** [comment]
It's not a 32bit lib, i'm pretty sure of it. Because I tried with an another lib so use cuda and it works, it was with jcuda libs ( http://www.jcuda.org/ ) and i got the right value.
The difference with sheepit and jcuda it's jna vs jni, they use jni and I'm using jna.

BUT the "funny" thing is I tried to create a jna lib who return a size_t value (like cuda's lib https://www.cs.cmu.edu/afs/cs/academic/class/15668-s11/www/cuda-doc/html/group__CUDA__DEVICE_gc6a0d6551335a3780f9f3c967a0fde5d.html ) and it works.

I know something is wrong somewhere but so far i can't tell where.
